### PR TITLE
TaskForm data class and tests

### LINKED ISSUE #83: APP, create task page, disable confirm button in time picker of end time if selected time is invalid.
Depending on the selected date range (if they are the same date), then the end time must be after the start time.
The suggestion is to disable the confirm button when selecting the end time, if the selected time is not after the start time.
Currently the "create task button" is disabled, if the end time is before the start time.

image

The confirm button in question is shown here:
image

Performing this is tedious since it would result in edge cases, what if the start time is 23:59? It is then impossible to select an end time.
Also the time picker is a component, and this mechanism would only need to be used in the end time picker.
Also, the data-flow would be messy, since it requires the start time, and the date ranges to be used in a function which is then passed to the end time picker.

### PR DESCRIPTION
**Pull request by henneboy**
Displays helpful error message and disables submit button if the form is not valid.
Next PR will have a .toTask() method which transforms it to the dto for the API.
Closes #83

### CODE REVIEW

**Review by S3JER — APPROVED** [review]

File: frontend/app/src/main/java/dk/scheduling/schedulingfrontend/pages/CreateTaskPage.kt [hunk 1]
```diff
@@ -72,62 +85,66 @@ fun CreateTaskPage(
...
+        }
 
         // Submit or Cancel
         Button(
             onClick = handleSubmission,
-            enabled = isValidInput(duration, datePickerValue.value, startTimeState, endTimeState),
+            enabled = task.status().isValid,
             modifier = modifier.fillMaxWidth(),
         ) {
             Text(text = "Create task")
```
Comment (Puvikaran2001) [inline frontend/app/src/main/java/dk/scheduling/schedulingfrontend/pages/CreateTaskPage.kt:150]: Can you instead add enabled parameter to filled button or outlined button?

**Review by Puvikaran2001 — CHANGES_REQUESTED** [review]

File: frontend/app/src/main/java/dk/scheduling/schedulingfrontend/model/Duration.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package dk.scheduling.schedulingfrontend.model
+
+data class Duration(val value: String) {
+    fun initializedAndInvalid(): Boolean {
```
Comment (Darkros1245) [inline frontend/app/src/main/java/dk/scheduling/schedulingfrontend/model/Duration.kt:4]:
```suggestion
    fun isInitializedAndInvalid(): Boolean {
```

**Review by Darkros1245 — CHANGES_REQUESTED** [review]

File: frontend/app/src/main/java/dk/scheduling/schedulingfrontend/model/Duration.kt (hunk 2 above)
Comment (henneboy) [reply frontend/app/src/main/java/dk/scheduling/schedulingfrontend/model/Duration.kt]: Har brugt renaming i editor til at gøre det så ikke at det breaker :)

File: frontend/app/src/main/java/dk/scheduling/schedulingfrontend/pages/CreateTaskPage.kt (hunk 1 above)
Comment (henneboy) [reply frontend/app/src/main/java/dk/scheduling/schedulingfrontend/pages/CreateTaskPage.kt]: Done

**Review by Darkros1245 — APPROVED** [review]

**Review by Puvikaran2001 — APPROVED** [review]
Nice

**Pull request merged**